c++/nda/simd/simd_cost.hpp

```diff
@@ -0,0 +1,222 @@
+#pragma once
+#include "../concepts.hpp"
+
+#include <type_traits>
+
+namespace nda {
+
+  //Forward declaration
+  template <typename ValueType, int Rank, typename Layout, char Algebra, typename ContainerPolicy>
```

Comment:
maybe create a header simd_fwd.h with all the forward declarations so that they are not in the way.